### DISCUSSION TITLE
Longer Ranking List

### DISCUSSION THREAD

**Discussion by BradKML** [analyticalmindsltd/smote_variants#61] [Ideas]
Currently requesting to see the longer rank list of the SMOTE competition as it seems interesting to see what other classification algorithms (with quadratic-like functions) can be injected e.g. RBF-based SVM, Naive Bayes, Gaussian Process, and QDA
- https://github.com/analyticalmindsltd/smote_variants/blob/master/docs/ranking.rst
- https://scikit-learn.org/stable/auto_examples/classification/plot_classifier_comparison.html

**Comment by gykovacs** [comment]
Not sharing the entire list is conceptual. Given the large number of oversamplers implemented, it is certain that there are bugs, misunderstandings, etc., and there were many more back when this study was created. Stating that an oversampler is the best (given our implementation) was found to be less harmful than stating that some oversampler is the worst among all (because this might be due to the misunderstanding of the algorithm, a typo in the paper from which we implemented, etc.). Therefore, intentionally, the lower end of the list is not disclosed.

On the other hand, as numerous bugfixes were released in the past months, possibly a completely new evaluation should be executed. Nevertheless, the ones on the top of the ranking are certainly reliable ones, however, there is a chance that some new ones would show up in the top.

**Reply by BradKML** [reply]
Thank you for the response, hope that re-evaluation can be done on a regular basis with more algorithms and evaluation metrics, as one would likely want to observe the top-30 algorithms and understand how each evaluation metrics may have different utility.

**Reply by gykovacs** [reply]
Yeah, actually, as there was only 1 oversampling technique added since the first release of the package, I did not find it crucial to reevaluate the performances. In the future, if somehow people would tend to add their oversamplers more regularly, this kind of regular evaluation could be added. Also, in the paper which is linked in the README you can find a bit longer lists and also rankings by the various characteristics of the datasets.
